### ISSUE TITLE
[feedback required] use newest vs historical display name in chat history

### ISSUE THREAD

**Issue by HarHarLinks** [vector-im/element-web#16419]
**Is your suggestion related to a problem? Please describe.**
This came up as a discussion in TWIM/TWIM offtopic continuation as a spin-off of #16262 and #16418.

Arguments for historical display names:
- ["historical state staying as-is is a feature in matrix, although it doesn't work in pills"](https://matrix.to/#/!xALORqBdeiSfgdrmUb:bpulse.org/$K-KOySE_t9FSgppQqHFRPKgNTiaTgX8LrpaeIqELA-M?via=bpulse.org&via=matrix.org&via=maunium.net), so represent it accurately
- historical accuracy: although using pills is optimal, references to display names can be made literally or indirectly. the reference is lost when the display name is replaced by a newer one
- user opinion: ["being able to see old names is kinda a cool novelty"](https://matrix.to/#/!xALORqBdeiSfgdrmUb:bpulse.org/$j_DifUCoASTuEfpERw1fzadIFrev2CKvcYGPZR8Xpoc?via=bpulse.org&via=matrix.org&via=maunium.net)
- subjectively can make it easier to track conversations
- more?

Arguments for newest display names:
- there are persons who want to have their name changed (and the old name gone) for serious reasons (e.g. trans people)
- user opinion: ["it makes it really Hard to follow a conversation if you have to remember every Name change of a Person to follow the conversation"](https://matrix.to/#/!xALORqBdeiSfgdrmUb:bpulse.org/$oM16vEcoMD0t9yY-7jB_Z6tCSw8obiuf1C0TK3wbB5Y?via=bpulse.org&via=matrix.org&via=maunium.net)
- more?

**Describe the solution you'd like**
Clearly removing all old names in favour of new ones isn't 100% perfectly possible due to plain text occurrences and more.

At this point, I explicitly **ask for feedback from people with need of having their name changed**, or expertise in this regard.
However, here are the results from my discussion in such a group:

- Add a toggle to the settings whether you prefer historical or current names. The same toggle can be presented inline on name change messages.
- Display the other name as tooltip on hover.
- (spec domain) Add a function that renames old display names to the new one, e.g. by overriding the above mentioned toggle when that user is seen, includes disabling displaying of old display names. If possible (probably not), purge the old names/rewrite respective events.

I do not see a reason to not have all of these implemented alongside each other, since they don't conflict.

"Show newest display names" would be the sensible default for the new setting, but having the toggle appear inline on name changes advertises it in an obvious manner.

Fallbacks (plain text) can only be display name at time of sending, or MXID. At this point in time MXID seems more sensible in addressing this issue (it's as close to the proposed default setting as possible). As a counterpoint one could argue that having a usable fallback is more important for recent messages and thus use the display name.

**Describe alternatives you've considered**

**Additional context**
- Whatever is chosen, pills and message headers need to be consistent: #16418
- MXID as fallbacks in fluffychat #16262

**Comment by t3chguy** [comment]
Duplicate of https://github.com/vector-im/element-web/issues/15763
